packages/api/tests/models/test-executions-model.js

```diff
@@ -139,225 +145,127 @@ test('generateRecord() returns a record with correct duration for non-running me
     status: 'running',
-    updatedAt: 123,
+    originalPayload: {},
     name: 'frank'
   };
 
   await executionModel.create(originalItem);
 
-  const updatedItem = {
-    ...originalItem,
-    updatedAt: 321,
-    name: 'joe'
-  };
-
-  const params = executionModel.buildDocClientUpdateParams(updatedItem);
-
-  await executionModel.dynamodbDocClient.update(params).promise();
+  cumulusMessage.meta.status = 'running';
+  await executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
 
   const fetchedItem = await executionModel.get({ arn: originalItem.arn });
 
   t.is(fetchedItem.status, 'running');
-  t.is(fetchedItem.updatedAt, 321);
+  t.deepEqual(fetchedItem.originalPayload, { value: 'my-payload' });
   t.is(fetchedItem.name, 'frank');
 });
 
-test.serial('buildDocClientUpdateParams() output can be used to create a new completed execution', async (t) => {
-  const { executionModel } = t.context;
+test.serial('storeExecutionFromCumulusMessage() can be used to create a new completed execution', async (t) => {
+  const { executionArn, executionModel, cumulusMessage } = t.context;
 
-  const item = {
-    arn: randomString(),
-    status: 'completed'
-  };
-
-  const params = executionModel.buildDocClientUpdateParams(item);
-
-  await executionModel.dynamodbDocClient.update(params).promise();
+  cumulusMessage.meta.status = 'completed';
+  await executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
 
-  const fetchedItem = await executionModel.get({ arn: item.arn });
+  const fetchedItem = await executionModel.get({ arn: executionArn });
 
   t.is(fetchedItem.status, 'completed');
 });
 
-test.serial('buildDocClientUpdateParams() output can be used to update a completed execution', async (t) => {
-  const { executionModel } = t.context;
+test.serial('storeExecutionFromCumulusMessage() can be used to update a completed execution', async (t) => {
+  const {
+    cumulusMessage,
+    executionArn,
+    executionModel,
+    executionName
+  } = t.context;
 
   const originalItem = {
-    arn: randomString(),
-    status: 'running',
-    updatedAt: 123,
+    arn: executionArn,
+    status: 'completed',
+    createdAt: 456,
     name: 'frank'
   };
 
   await executionModel.create(originalItem);
 
-  const updatedItem = {
-    ...originalItem,
-    status: 'completed',
-    updatedAt: 321,
-    name: 'joe'
-  };
-
-  const params = executionModel.buildDocClientUpdateParams(updatedItem);
-
-  await executionModel.dynamodbDocClient.update(params).promise();
+  cumulusMessage.meta.status = 'completed';
+  await executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
 
-  const fetchedItem = await executionModel.get({ arn: originalItem.arn });
+  const fetchedItem = await executionModel.get({ arn: executionArn });
 
   t.is(fetchedItem.status, 'completed');
-  t.is(fetchedItem.updatedAt, 321);
-  t.is(fetchedItem.name, 'joe');
+  t.is(fetchedItem.createdAt, 123);
+  t.is(fetchedItem.name, executionName);
```

Comment:
This is an interesting test case. Of course the `arn` is the key, but this feels wrong, though I know it's logically right. I guess it's just weirding me out because I have trouble imagining a scenario where the execution name changes but the `arn` does not.

Reply:
Good point. I'll update the test to use a less confusing condition

Reply:
dedcfa5e2974e03bb8b939eb3cfaa62ce39b20b3